### PR TITLE
Sync tests for practice exercise ``ledger``

### PR DESCRIPTION
**Pull request by manumafe98**
# pull request

This issue addresses: [#2388](https://github.com/exercism/java/issues/2388)

The goal is to sync the tests of the ledger practice exercise

Reviewer Resources:

[Track Policies](https://github.com/exercism/java/blob/main/POLICIES.md#event-checklist)

### CODE REVIEW

**Comment by manumafe98** [comment]
@sanderploegsma do you know what can be the issue here? Because i'm running the checks locally and seem fine
````
Manumafe@DESKTOP-GUDFCER MINGW64 /e/repos/java/exercises/practice/ledger (sync-tests-ledger)
$ gradle check

> Task :practice:ledger:compileJava
  (source = E:\repos\java\exercises\practice\ledger\.meta\src\reference\java\Ledger.java)

> Task :practice:ledger:compileTestJava
  (source = E:\repos\java\exercises\practice\ledger\build\gen\test\java\LedgerTest.java)

> Task :practice:ledger:test

LedgerTest > dutchLocaleAndEuros PASSED

LedgerTest > emptyLedgerUS PASSED

LedgerTest > creditAndDebit PASSED

LedgerTest > euros PASSED

LedgerTest > dutchNegativeNumberWith3DigitsBeforeDecimalPoint PASSED

LedgerTest > americanNegativeNumberWith3DigitsBeforeDecimalPoint PASSED

LedgerTest > finalOrderTieBreakerIsChange PASSED

LedgerTest > dutchLocale PASSED

LedgerTest > multipleEntriesOnSameDateOrderedByDescription PASSED

LedgerTest > overlongDescriptions PASSED

LedgerTest > oneEntry PASSED

BUILD SUCCESSFUL in 5s
6 actionable tasks: 6 executed
````

**Comment by sanderploegsma** [comment]
You can download the test result artifacts from the GitHub Actions workflow screen to see what is going on. I took a look and found the following test failures. Hope that helps!

<table>
<thead>
<tr>
<th>Test</th>
<th>Error</th>
</tr>
<tr>
<td>

`dutchLocaleAndEuros`

</td>
<td>

```text
Message: Index 1 out of bounds for length 1
Exception: java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
	at Ledger.format(Ledger.java:110)
	at LedgerTest.dutchLocaleAndEuros(LedgerTest.java:152)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
```

</td>
</tr>

<tr>
<td>

`creditAndDebit`

</td>
<td>

```text
Message: Index 1 out of bounds for length 1
Exception: java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
	at Ledger.format(Ledger.java:112)
	at LedgerTest.creditAndDebit(LedgerTest.java:55)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
```

</td>
</tr>

<tr>
<td>

`euros`

</td>
<td>

```text
Message: Index 1 out of bounds for length 1
Exception: java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
	at Ledger.format(Ledger.java:112)
	at LedgerTest.euros(LedgerTest.java:122)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
```

</td>
</tr>

<tr>
<td>

`dutchNegativeNumberWith3DigitsBeforeDecimalPoint`

</td>
<td>

```text
Message: Index 1 out of bounds for length 1
Exception: java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
	at Ledger.format(Ledger.java:110)
	at LedgerTest.dutchNegativeNumberWith3DigitsBeforeDecimalPoint(LedgerTest.java:167)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
```

</td>
</tr>

<tr>
<td>

`americanNegativeNumberWith3DigitsBeforeDecimalPoint`

</td>
<td>

```text
Message: Index 1 out of bounds for length 1
Exception: java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
	at Ledger.format(Ledger.java:112)
	at LedgerTest.americanNegativeNumberWith3DigitsBeforeDecimalPoint(LedgerTest.java:182)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
```

</td>
</tr>

<tr>
<td>

`finalOrderTieBreakerIsChange`

</td>
<td>

```text
Message: Index 1 out of bounds for length 1
Exception: java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
	at Ledger.format(Ledger.java:112)
	at LedgerTest.finalOrderTieBreakerIsChange(LedgerTest.java:90)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
```

</td>
</tr>

<tr>
<td>

`dutchLocale`

</td>
<td>

```text
Message: Index 1 out of bounds for length 1
Exception: java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
	at Ledger.format(Ledger.java:110)
	at LedgerTest.dutchLocale(LedgerTest.java:137)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
```

</td>
</tr>

<tr>
<td>

`multipleEntriesOnSameDateOrderedByDescription`

</td>
<td>

```text
Message: Index 1 out of bounds for length 1
Exception: java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
	at Ledger.format(Ledger.java:112)
	at LedgerTest.multipleEntriesOnSameDateOrderedByDescription(LedgerTest.java:72)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
```

</td>
</tr>

<tr>
<td>

`overlongDescriptions`

</td>
<td>

```text
Message: Index 1 out of bounds for length 1
Exception: java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
	at Ledger.format(Ledger.java:112)
	at LedgerTest.overlongDescriptions(LedgerTest.java:107)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
```

</td>
</tr>

<tr>
<td>

`oneEntry`

</td>
<td>

```text
Message: Index 1 out of bounds for length 1
Exception: java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
	at Ledger.format(Ledger.java:112)
	at LedgerTest.oneEntry(LedgerTest.java:40)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
```

</td>
</tr>
</table>

**Comment by manumafe98** [comment]
Okey so for some reason this line https://github.com/exercism/java/blob/37bbd1bb453189e7fc5fb5596a1513edbfcc9da1/exercises/practice/ledger/.meta/src/reference/java/Ledger.java#L97

Locally was breaking all my tests, so I changed it to comma instead of dot, but here breaks with comma and works with dot. Its weird, do you know what could be causing this @sanderploegsma ?

File: exercises/practice/ledger/src/main/java/Ledger.java [hunk 1]
```diff
@@ -106,12 +106,23 @@ public String format(String cur, String loc, LedgerEntry[] entries) {
...
+                if (e.getChange() < 0 && loc.equals("en-US")) {
+                    amount = "(" + amount + ")";
+                } else if (e.getChange() < 0 && loc.equals("nl-NL")) {
+                    amount = curSymb + " -" + amount.replace(curSymb, "").trim() + " ";
+                } else if (loc.equals("nl-NL")) {
+                    amount = " " + amount + " ";
+                } else {
+                    amount = amount + " ";
+                }
+
```
Comment (sanderploegsma) [inline exercises/practice/ledger/src/main/java/Ledger.java:125]: Due to the large diff in `LedgerTest.java` I'm not 100% sure what has changed in the behavior of this exercise, can you explain why these changes were needed?

Comment (manumafe98) [reply exercises/practice/ledger/src/main/java/Ledger.java:125]: Yes! well first of all this is a reformat exercise, so I have to match the main code that receives the user with the reference to pass the new tests.
The new synced tests changed:
- They pass an ``int`` instead of a ``double``
- They asked for a different format of negative values in ``en-US`` ``($10.00)``
- Also if the number is positive to always have a space after the number
- And for ``nl-NL`` a space between the currency and the number

**Review by sanderploegsma — APPROVED** [review]
